internal/store/utils.go

```diff
@@ -75,6 +75,10 @@ func kubeLabelsToPrometheusLabels(labels map[string]string) ([]string, []string)
 }
 
 func kubeAnnotationsToPrometheusLabels(annotations map[string]string) ([]string, []string) {
+	// Remove Last applied configuration annotation since it can potentially store sensitive data.
+	// Refer to https://github.com/kubernetes/kube-state-metrics/issues/854 for more info.
+	// TODO: We need to rethink how we expose annotations since it's pretty unbounded.
+	delete(annotations, "kubectl_kubernetes_io_last_applied_configuration")
```

Comment:
This should actually be `"kubectl.kubernetes.io/last-applied-configuration"` because we don't yet transform it to the Prometheus label format yet. We could also just take the constant `v1.LastAppliedConfigAnnotation` here to make it clear. https://github.com/kubernetes/api/blob/d0d4f3afa3ab/core/v1/annotation_key_constants.go#L71:2

Reply:
Done :). Thanks.